sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -131,6 +131,7 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       df.explode('letters) {
         case Row(letters: String) => letters.split(" ").map(Tuple1(_)).toSeq
       }
+    assert(!df2.queryExecution.toString.contains("!"))
```

Comment:
@cloud-fan I like your suggestion!

`resolved` is not defined in `SparkPlan`. It is only available in `LogicalPlan`. I am not sure if you want me to define it in `SparkPlan` and overwrite it if necessary like what we did in `LogicalPlan`?

Thank you!